### PR TITLE
fixes params groups

### PR DESCRIPTION
**Pull request by wyli**
### Description
fixes issues of empty generators in generate_param_groups

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/optimizers/utils.py [hunk 1]
```diff
@@ -48,27 +50,31 @@ def generate_param_groups(
...
+        elif ty.lower() == "filter":
             layer_params = filter(func, network.named_parameters())
+            layer_ids = filter(func, network.named_parameters())
         else:
-            raise ValueError(f"unsuppoted layer match type: {ty}.")
+            raise ValueError(f"unsupported layer match type: {ty}.")
 
         params.append({"params": layer_params, "lr": lr})
-        _layers.extend(list(map(id, layer_params)))
+        _layers.extend(list(map(id, layer_ids)))
```
Comment (Nic-Ma) [inline monai/optimizers/utils.py:77]: Can we change to `layer_params.deepcopy()` to avoid duplicated `func` and `filter` calls?

Thanks.

**Review by Nic-Ma — APPROVED** [review]

File: monai/optimizers/utils.py (hunk 1 above)
Comment (wyli) [reply monai/optimizers/utils.py]: thanks, I added local functions to return generators, that would work like making fresh copies.

**Review by Nic-Ma — APPROVED** [review]
Looks good to me.
Thanks for your quick update.

**Pull request merged**